src/pages/Signup.tsx

```diff
@@ -0,0 +1,239 @@
+import React, { useState } from "react";
+import { withRouter, RouteComponentProps } from "react-router-dom";
+import { useCookies } from "react-cookie";
+import md5 from "md5";
+
+import { API_DOMAIN } from "../config";
+
+import { Button, Input, Layout, Divider, Typography } from "antd";
+
+import Header from "../components/Header";
+import "../styles/Signup.css";
+
+const ALL_FIELDS_SET = "All fields must be filled before proceeding.";
+const PASSWORD_MATCH = "Password should match.";
+const UNAVAILABLE_USERNAME = "Username is taken. Please enter a new one.";
+
+const Signup: React.FC<RouteComponentProps> = (props) => {
+  const [cookies, setCookie] = useCookies(["accessToken"]);
+  const [email, setEmail] = useState("");
+  const [username, setUsername] = useState("");
+  const [password, setPassword] = useState("");
+  const [confirmPassword, setConfirmPassword] = useState("");
+
+  const [fullName, setFullname] = useState("");
+  const [institution, setInstitution] = useState("");
+  const [dept, setDept] = useState("");
+  const [position, setPosition] = useState("");
+
+  const [error, setError] = useState("");
+
+  const onCreateClick = async (): Promise<void> => {
+    if (
+      !username &&
```

Comment:
these should all be OR instead of AND, as the error does not pop up correctly :)